internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -967,6 +994,15 @@ func (r *AgentServiceConfigReconciler) getOpenshiftVersions(log logrus.FieldLogg
 	return string(encodedVersions)
 }
 
+func (r *AgentServiceConfigReconciler) getCMHash(ctx context.Context, namespacedName types.NamespacedName) (string, error) {
+	cm := &corev1.ConfigMap{}
+	if err := r.Get(ctx, namespacedName, cm); err != nil {
+		r.Log.Error(err, "Failed to get configmap", "NamespacedName", namespacedName)
+		return "", err
+	}
+	return checksumMap(cm.Data)
```

Comment:
What do you think (for safety measures) about logging errors coming from `checksumMap` and ignore them?

Maybe I'm being too paranoid but the scenario I would like to prevent is that some unhandled encoding issue (I realize that everything is b64 in this function) or some other internal problem, could cause this function to fail and, therefore, all the steps after it.